source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,143 @@
+==============================
+Implementing a DataManipulator
+==============================
+
+If you want to properly implement a ``DataManipulator`` you have to:
+
+1. Implement the ``DataManipulator`` itself
+#. Implement the ``ImmutableDataManipulator``
+
+Upon completion of the first steps you'll have to get this done too:
+
+#. Register the ``Key`` in the ``KeyRegistry``
+#. Implement the ``DataProcessor``
+#. Implement the ``ValueProcessor`` for each value being represented by the ``DataManipulator``
+
+1. Implementing the DataManipulator
+===================================
+
+Implementing the ``DataManipulator`` is done by extending AbstractData in the easiest case. Heres a short example:
+
+.. code-block:: java
+
+  public class SpongeFlyingData extends AbstractData<FlyingData, ImmutableFlyingData> implements FlyingData {
+ /* your code here */
+  }
+
+Let's assume we have a single value based manipulator, like for example the ``flying`` state on an entity which can be ``true`` or ``false``.
+We abstracted things above, so now we're able to just extend ``AbstractBooleanData``.
+The next step is to provide the generic ``DataManipulator`` that follows the generic of ``FlyingData``. In this case we use ``FlyingData.class`` in our ``super`` constructor,
+followed by the ``Key`` which identifies the recommended ``value`` of the flying state and last but not least the ``Key`` itself: ``Keys.IS_FLYING``.
```

Comment:
Should i stick with ``KEYS.IS_FLYING`` or go with ``KEYS.FLYING``? I just used the name currently used in FlyingData implementation.